### PR TITLE
Hepmc reader interface

### PR DESCRIPTION
**Pull request by kkauder**
Can read all file formats supported by hepmc3's deduce_reader, https://gitlab.cern.ch/hepmc/HepMC3/-/blob/master/include/HepMC3/ReaderFactory.h

Turned on by default, as long as a HepMC3 library is found. A small example file is in data/neutral_current.hepmc3

### CODE REVIEW

**Comment by kkauder** [comment]
There shouldn't be much of an issue, but if you have a moment, please take a quick look. Otherwise I'll merge tonight.

**Comment by makotoasai** [comment]
Looks fine with me.

File: Core/run.mac [hunk 1]
```diff
@@ -22,25 +22,25 @@
...
 #######################################################
 # Detector components
 #######################################################
-/eAST/component/beampipe 1
-/eAST/component/beampipe/envelopeGdmlFile ../Components/Beampipe/data/Detector_chamber_210222_BG-05-21-2021_single_volume.gdml
-/eAST/component/beampipe/gdmlFile ../Components/Beampipe/data/Detector_chamber_210222_test1_BG-04-07-2021.gdml
-/eAST/component/beampipe/materialFile ../Components/Beampipe/data/Detector_chamber_210222_test1_BG-04-07-2021.mat
-/eAST/component/beampipe/rotate x -90. deg
-/eAST/component/beampipe/rotate y -90. deg
+# /eAST/component/beampipe 1
```
Comment (wdconinc) [inline Core/run.mac:38]: Shouldn't we keep this enabled? Even if we know it doesn't work on your setup?

**Review by wdconinc — APPROVED** [review]
Looks good to me.

File: Core/run.mac (hunk 1 above)
Comment (kkauder) [reply Core/run.mac:38]: Fair question, I wasn't sure what to do. I don't want a user to try and test the hepmc interface and than run into obscure and mostly unrelated problems. But I'll reenable it and add some comments in the macro.

**Pull request merged**